core/templates/pages/topic-viewer-page/story-node-tile/story-node-tile.component.ts

```diff
@@ -0,0 +1,243 @@
+    private windowDimensionsService: WindowDimensionsService
+  ) {}
+
+  ngOnInit(): void {
+    this.thumbnailFilename = this.node.getThumbnailFilename() as string;
+    this.thumbnailBgColor = this.node.getThumbnailBgColor() as string;
+    this.nodeId = this.node.getId();
+    this.pathIconParameters = this.generatePathIconParameters();
+    this.explorationId = this.node.getExplorationId() as string;
+    let states: StateObjectsBackendDict;
+    this.readOnlyExplorationBackendApiService
+      .loadLatestExplorationAsync(
+        this.explorationId,
+        this.urlService.getPidFromUrl()
+      )
+      .then(response => {
+        states = response.exploration.states;
+
+        this.mostRecentlyReachedCheckpoint =
+          response.most_recently_reached_checkpoint_state_name;
+
+        let exploration = this.explorationObjectFactory.createFromBackendDict({
+          auto_tts_enabled: response.auto_tts_enabled,
+          draft_changes: [],
+          is_version_of_draft_valid: true,
+          init_state_name: response.exploration.init_state_name,
+          param_changes: response.exploration.param_changes,
+          param_specs: response.exploration.param_specs,
+          states: response.exploration.states,
+          title: response.exploration.title,
+          draft_change_list_id: response.draft_change_list_id,
+          language_code: response.exploration.language_code,
+          version: response.version,
+          next_content_id_index: response.exploration.next_content_id_index,
+          exploration_metadata: response.exploration_metadata,
+        });
+        this.explorationEngineService.exploration = exploration;
+        this.prevSessionStatesProgress =
+          this.explorationEngineService.getShortestPathToState(
+            states,
+            this.mostRecentlyReachedCheckpoint
+          );
+        for (let i = 0; i < this.prevSessionStatesProgress.length; i++) {
+          // Set state name of a previously completed state.
+          let stateName = this.prevSessionStatesProgress[i];
+          if (this.mostRecentlyReachedCheckpoint === stateName) {
+            break;
+          }
+          this.completedcheckpoints += 1;
+        }
+        this.nodeDataLoaded = true;
+        this.getCheckpointCount();
+        this.getCompletedProgressStatus();
+      });
+  }
+
+  async getCheckpointCount(): Promise<void> {
+    return this.readOnlyExplorationBackendApiService
+      .fetchExplorationAsync(this.explorationId, null)
+      .then((response: FetchExplorationBackendResponse) => {
+        this.expStates = response.exploration.states;
+        let count = 0;
+        for (let [, value] of Object.entries(this.expStates)) {
+          if (value.card_is_checkpoint) {
+            count++;
+          }
+        }
+        this.checkpointCount = count;
+        this.totalCheckpointLoaded = true;
+        this.getCompletedProgressStatus();
+      });
+  }
+
+  getCompletedProgressStatus(): void {
```

Comment:
The function name is getCompletedProgressStatus but the function does not return anything, can we rename this?